server/lib/util.py

```diff
@@ -868,8 +835,63 @@ def _get_highest_coverage_date(observation_entity_counts_by_date,
           ],
               key=lambda item: item['count'])['count']
   } for obs in observation_dates]
-  best_coverage = max(date_counts, key=lambda date_count: date_count['count'])
-  return best_coverage['date']
+  return date_counts
+
+
+def fetch_highest_coverage(variables: List[str],
+                           all_facets: bool,
+                           entities: List[str] | None = None,
+                           parent_entity: str | None = None,
+                           child_type: str | None = None,
+                           facet_ids: List[str] | None = None):
+  """
+  Fetches the latest available data with the best coverage for the given a
+  (list of entities OR (parent entity and child type)), variables, and facets.
+  If multiple variables are passed in, selects dates with the overall highest
+  coverage among all variables.
+
+  Response format:
+  {
+    "facets": {
+      <facet_id>: {<facet object>}
+    },
+    "data": {
+      <var_dcid>: {
+        <entity_dcid>: {
+          <observation point(s)>
```

Comment:
nit: maybe mention that multiple observation points returned when all facets is true & only one returned if all facets is false

Reply:
done